exodus_core/analysis/static_analysis.py

```diff
@@ -168,27 +168,17 @@ def get_embedded_classes(self):
         if self.classes is not None:
             return self.classes
 
-        class_regex = re.compile(r'classes.*\.dex')
-        with TemporaryDirectory() as tmp_dir:
-            with zipfile.ZipFile(self.apk_path, "r") as apk_zip:
-                class_infos = (info for info in apk_zip.infolist() if class_regex.search(info.filename))
-                for info in class_infos:
-                    apk_zip.extract(info, tmp_dir)
-            dexdump = which('dexdump')
-            cmd = '{} {}/classes*.dex | perl -n -e\'/[A-Z]+((?:\w+\/)+\w+)/ && print "$1\n"\'|sort|uniq'.format(
-                dexdump, tmp_dir)
-            try:
-                self.classes = subprocess.check_output(
-                    cmd,
-                    stderr=subprocess.STDOUT,
-                    shell=True,
-                    universal_newlines=True
-                ).splitlines()
-                logging.debug('{} classes found in {}'.format(len(self.classes), self.apk_path))
-                return self.classes
-            except subprocess.CalledProcessError:
-                logging.error('Unable to decode {}'.format(self.apk_path))
-                raise Exception('Unable to decode the APK')
+        dexs = list(APK(filename=self.apk_path).get_all_dex())
+
+        class_names = []
+        for dex in dexs:
+            d = DalvikVMFormat(dex)
+            dex_class_names = d.get_classes_names()
+            for c in dex_class_names:
+                if "$" not in c:
+                    class_names.append(c[1:-1])
```

Comment:
Not sure to understand the c[1:-1]

Reply:
@simpnu the name of the classes start with `L` and finish with `;`. For example `Lcom/google/common/cache/LocalCache;`. So I removed these characters to keep only the path and the name of the class.